ssl/statem/statem_lib.c

```diff
@@ -2415,3 +2415,162 @@ int tls13_restore_handshake_digest_for_pha(SSL *s)
     }
     return 1;
 }
+
+#if !defined(OPENSSL_NO_COMP)
+
+/* Intended to be used after WPACKET_reserve_bytes() */
+static int wpacket_advance_write(WPACKET *pkt, size_t len)
```

Comment:
Isn't this just `WPACKET_allocate_bytes`?

Reply:
`WPACKET_allocate_bytes()` allocates memory, this is used *after* `WPACKET_reserve_bytes()` to advance the write pointer, without allocating. Presumably, nothing would be allocated (since that's already been done).
I made the change; all the tests still work.